src/main/java/org/javamodularity/moduleplugin/tasks/CompileTask.java

```diff
@@ -43,13 +43,9 @@ private void configureCompileJava(JavaCompile compileJava) {
                         boolean hasKt = helper().kotlinMainSourceDirectorySet().getFiles().stream()
                                 .map(File::getName)
                                 .anyMatch(name -> name.endsWith(".kt"));
-                        boolean hasJava = helper().mainSourceSet().getJava().getFiles().stream()
-                                .map(File::getName)
-                                .filter(name -> ! name.equals("module-info.java"))
-                                .anyMatch(name -> name.endsWith(".java"));
-                        if(hasKt && !hasJava) {
+                        if(hasKt) {
```

Comment:
I believe calling `setCompileModuleInfoSeparately(true)` as well as checking for any Kotlin files may be redundant here.

I'd think that we could just call `moduleOptions.addModuleClassesDir(compileKotlin.getDestinationDir())` whenever `compileKotlin` task is present. Even if there are no main Kotlin sources, it does no harm, but we save all those extra checks that add complexity to the codebase of this plugin.

It's different with Groovy, because with Groovy, we *need* to call `setCompileModuleInfoSeparately(true)`, and this should truly be done only when necessary (i.e. when there are any main Groovy sources).

Reply:
Fixed.